src/map.react.js

```diff
@@ -303,6 +305,20 @@ export default class MapGL extends Component {
     map.addSource(update.id, newSource);
   }
 
+  // Hover and click only query layers whose interactive property is true
```

Comment:
I think more documentation here would be very helpful for casual maintainers/readers. 
1) You could turn this into a full JSDoc section.
2) You could add explanatory text by rewriting the great summary you did for this PR. Maybe add the link to the API doc

```
https://www.mapbox.com/mapbox-gl-js/api/#Map#queryRenderedFeatures
```